Add issue/PR templates, CoC, CONTRIBUTING.md

chriszs opened the pull request:
## Description
This PR adds the issue and pull request templates and docs so we can have a centralized definition of these.

## Related Issues
Resolves #1 
Resolves #3

chriszs committed:
Add issue/PR templates

chriszs commented on pull_request_template.md:
```diff
@@ -0,0 +1,27 @@
+- [ ] Documentation
+- [ ] Consolidated commits
+- [ ] Relevant labels assigned
+- [ ] Changelog(s) updated
```
`Changelog(s) updated` is the only part of this template that seems like it should vary depending on the repo. Should I leave this out?

slifty replied:
I wouldn't mind adopting a habit of adding a changelog to all projects (monorepo or not), this is a good enough excuse for that.

chriszs committed:
Add CoC/contributing.md

This adds the CoC and CONTRIBUTING.md
sourced from the tv-kitchen repo
with some light edits mostly related to
relative links and references in
CONTRIBUTING.md.

#3 Add CoC/contributing.md

chriszs commented:
[Another consideration.](https://github.com/tvkitchen/appliances/issues/17) But you can still overwrite templates locally even if they have central definition.

chriszs commented:
Okay, so here's the situation: I started work on a separate PR that added the CoC and CONTRIBUTING.md files to this repo as per #3. That's a little more involved because CONTRIBUTING in particular includes a bunch of relative references to other Markdown files and sections that might be specific to a repo, like Sequelize migration instructions. It's a little unclear how those references are going to work in GitHub and it might involve changing them. The plan was to mark some of those up and work through how to resolve them. But I accidentally committed those to this branch instead of a new one.

Before I could fix it, someone turned off my access to push branches to this repo. So I'm going to pretend I always meant to do that in this PR also.

This has some utility, because the tvkitchen repo lacks a bug template, website and meta lack docs (see this issue in [website](https://github.com/tvkitchen/tv-kitchen-website/issues/1)). Merging this would fix that globally.

chriszs commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,175 @@
+# Contribution Guidelines
+
+This document contains information of use to developers looking to improve this codebase.  See [README.md](https://github.com/tvkitchen/tv-kitchen/blob/master/README.md) for an introduction to this project. All discussions about this project fall under our [code of conduct](CODE_OF_CONDUCT.md).
```
Changed this to refer to the project, link to the README in the tvkitchen repo.

chriszs commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,175 @@
+
+Please submit changes via pull request, even if you have direct commit access to the repository.  The PR process allows us to get additional eyes on change proposals, and ensures that your changed code [builds cleanly via Travis CI's automated builds](https://travis-ci.org/tvkitchen/tv-kitchen/).
+
+As you work on your branch, try to test it locally to ensure that it still builds and deploys properly, and that you haven't introduced new accessibility bugs. See `docs/TESTING.md` for more instructions.
```
May need to either put TESTING.md in the same repo or change this so it's no longer a link. Unclear whether GitHub has some magic way of handling this. Should check.

chriszs commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,175 @@
+
+As you work on your branch, try to test it locally to ensure that it still builds and deploys properly, and that you haven't introduced new accessibility bugs. See `docs/TESTING.md` for more instructions.
+
+Generally, the more controversial, complex or large a change, the more opportunity people should have to comment on it.  That means it should garner more comments/approvals, or it means it should sit longer before being merged. You can talk with us about a change you'd like to make before or while you work on it; see [the participation section of the README](README.md#participating). We don't hav ...
```
Another relative link.

chriszs commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,175 @@
+
+As to when to merge, that's a judgment call.  Usually once an "approved" review goes through, and there aren't any more changes requested, then the author of the PR will merge it (if they have access to push to master).  Generally, wait to merge until the conversation around a change has concluded.  If you're unsure, ask!  "Is this ready to merge?" is often a useful next step in the conversation.
+
+If your PR fixes a bug or adds a feature, please write a test to go with the change (see [TESTING.md](docs/TESTING.md) for details on our testing framework).  If the change involves libraries or would be difficult to test, please use a Given-When-Then description or describe in your PR message how reviewers should test that your change works as expected.
```
Same.

chriszs commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,175 @@
+
+### Document as you go
+
+If you introduce a new framework or dependency, add the necessary information to [INSTALL.md](docs/INSTALL.md) and the other documentation.
```
Another relative link.

chriszs commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,175 @@
+
+### Migrations
+
+The golden rule of migrations: Never edit a migration file that has been committed to master.
```
Are Sequelize migration instructions applicable globally?

slifty replied:
I think we could keep this but make it more general -- "For projects with migrations...."

chriszs commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,175 @@
+
+## Thank you!
+
+The [Corporate Overlords](https://biffud.com/people) of [Bad Idea Factory](https://biffud.com/) appreciate your dedication and service to the public good.
```
Added some friendly, explanatory links to BIF's website here.

reefdog commented:
FWIW I think it's perfectly fine to remove the links if we can't wrestle them into relative submission. It's Very Not Hard for someone to just look up `TESTING.md` wherever it is in the repo.

chriszs commented:
Makes sense to me.

slifty commented:
Wait how did you lose access to push branches (I did not change anything!)

slifty commented:
I agree with both of you (I think you both agree, right?) -- and we can always add links back in later after shifting over to this model.

slifty commented:
Gonna merge these in, pray for mojo...
